src/accountsdb/manager.zig

```diff
@@ -20,260 +20,201 @@ const AccountFile = sig.accounts_db.accounts_file.AccountFile;
-    pub fn run(self: *Manager, allocator: Allocator) !void {
-        const zone = tracy.Zone.init(@src(), .{ .name = "accountsdb runManagerLoop" });
-        defer zone.deinit();
-
-        self.timer = try std.time.Timer.start();
-        while (!self.config.exit.load(.acquire)) {
-            try self.manage(allocator);
-            const elapsed = self.timer.lap();
-            if (elapsed < DB_MANAGER_LOOP_MIN.asNanos()) {
-                const delay = DB_MANAGER_LOOP_MIN.asNanos() - elapsed;
-                std.time.sleep(delay);
-            }
-        }
+    var failed: bool = false;
+    defer if (!failed) {
+        db.logger.info().logf("successfully flushed slot {} to disk", .{newly_rooted_slot});
+        tracy.frameMarkNamed("rooted slots flushed");
+    };
+    errdefer |err| {
+        db.logger.err().logf(
+            "failed to flushed slot {} to disk, err: {}",
+            .{ newly_rooted_slot, err },
+        );
+        failed = true;
     }
 
-    pub fn manage(self: *Manager, allocator: Allocator) !void {
-        tracy.frameMarkNamed("manager loop");
+    // make a new AccountFile for our newly rooted slot, and "flush" the data to it
+    const file_id = new_id: {
+        // flush fail = loss of account data on slot -- should never happen
+        errdefer |err| db.logger.err()
+            .logf("flushSlot {d} error: {s}", .{ newly_rooted_slot, @errorName(err) });
 
-        {
-            const unrooted_accounts, var unrooted_accounts_lg =
-                self.db.unrooted_accounts.readWithLock();
-            defer unrooted_accounts_lg.unlock();
-
-            // we're careful to load this value only after acquiring a read lock on the
-            // account cache, such as to avoid the edge case where:
-            // * we load the largest rooted slot value.
-            // * the account cache lock is acquired by a different thread.
-            // * another thread increments the largest rooted slot value while
-            //   we're waiting on the account cache lock.
-            // * we eventually acquire the lock, but have already read a now-stale
-            //   largest rooted slot value.
-            const root_slot = self.db.largest_rooted_slot.load(.monotonic);
-
-            // flush slots <= root slot
-            // TODO: account for forks when consensus is implemented
-            var unrooted_slot_iter = unrooted_accounts.keyIterator();
-            while (unrooted_slot_iter.next()) |unrooted_slot| {
-                if (unrooted_slot.* <= root_slot) {
-                    // NOTE: need to flush all references <= root_slot before we call clean
-                    // or things break by trying to clean unrooted references
-                    // NOTE: this might be too much in production, not sure
-                    try self.flush_slots.append(allocator, unrooted_slot.*);
-                }
-            }
-        }
+        break :new_id try flushSlot(db, newly_rooted_slot);
```

Comment:
What happens if the caller attempts to root a slot <= the highest rooted slot? Previously this was checked because the manager didn't know when it was notified about a rooted slot. But now I think you've baked in an assumption that the caller will never do this. So what's the behavior if the caller does that? Maybe we should have a check and:
- return error if the slot is less than the last flushed slot
- either do nothing or return error if the slot is equal to the last flushed slot